bundler/lib/bundler/dsl.rb

```diff
@@ -447,8 +447,25 @@ def check_path_source_safety
     end
 
     def check_rubygems_source_safety
-      return unless @sources.aggregate_global_source?
+      if @sources.implicit_global_source?
+        implicit_global_source_warning
 
+        return
```

Comment:
I know I did suggest to early return here, but looking at the final version, an early return usually suggests and exceptional condition, whereas in this case both warnings are "simmetrical" (of equal importance). So I think an `elsif` clause is a better style fit. So

```ruby
      if @sources.implicit_global_source?
        implicit_global_source_warning
      elsif @sources.aggregate_global_source?
        multiple_global_source_warning
      end
```